### PR TITLE
Added invite option to room's context menu

### PR DESCRIPTION
**Pull request by jaiwanth-v**
Added invite option to room's context menu

ezgif com-gif-maker (2)

Signed-off-by: Jaiwanth <jaiwanth2011@gmail.com>

### CODE REVIEW

File: src/components/views/rooms/RoomTile.tsx [hunk 1]
```diff
@@ -470,7 +481,11 @@ export default class RoomTile extends React.PureComponent<IProps, IState> {
                         label={lowPriorityLabel}
                         iconClassName="mx_RoomTile_iconArrowDown"
                     />
-
+                    <IconizedContextMenuOption
+                        onClick={this.onInviteClick}
+                        label={_t("Invite to this room")}
+                        iconClassName="mx_RoomTile_iconInvite"
+                    />
```
Comment (t3chguy) [inline src/components/views/rooms/RoomTile.tsx:488]: this should probably check whether the user has permission to complete that action

Comment (jaiwanth-v) [reply src/components/views/rooms/RoomTile.tsx:488]: Oh okay will do that

Comment (jaiwanth-v) [reply src/components/views/rooms/RoomTile.tsx:488]: Is there any code formatter that can be used for this project. I am used to them and can't find anything which works well with this project's code.

Comment (SimonBrandner) [reply src/components/views/rooms/RoomTile.tsx:488]: @jaiwanth-v, I use [this](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) extension for VS Code but it's not perfect. For other questions like this, I would recommend the [#element-dev](https://matrix.to/#/#element-dev:matrix.org?via=matrix.org&via=vector.modular.im&via=mozilla.org) room

Comment (jaiwanth-v) [reply src/components/views/rooms/RoomTile.tsx:488]: Thanks👍

**Comment by jaiwanth-v** [comment]
ezgif com-gif-maker (3)

**Comment by jaiwanth-v** [comment]
Sorry, I removed review request by mistake

**Comment by SimonBrandner** [comment]
@jaiwanth-v, could you please change the description so there isn't `fixes <issue number>`, so that this doesn't close the whole issue

**Comment by jaiwanth-v** [comment]
Okay

File: src/components/views/rooms/RoomTile.tsx [hunk 2]
```diff
@@ -451,6 +462,16 @@ export default class RoomTile extends React.PureComponent<IProps, IState> {
             const isLowPriority = roomTags.includes(DefaultTagID.LowPriority);
             const lowPriorityLabel = _t("Low Priority");
 
+            const inRoom = this.props.room && this.props.room.getMyMembership() === "join";
+            const userId = MatrixClientPeg.get().getUserId();
+            let canInvite = inRoom;
+            const powerLevels = this.props.room.currentState
```
Comment (t3chguy) [inline src/components/views/rooms/RoomTile.tsx:468]: seems a bit strange that you're checking for the truthiness of `this.props.room` above but not here.

the `IProps` say it should always be set

**Review by t3chguy — APPROVED** [review]
Code wise looks good other than the small nit where `this.props.room && ` can be omitted.

Pending design review to determine if it is something we want

File: src/components/views/rooms/RoomTile.tsx (hunk 2 above)
Comment (jaiwanth-v) [reply src/components/views/rooms/RoomTile.tsx:468]: Yes, I've changed that.

**Comment by niquewoodhouse** [comment]
Hi @jaiwanth-v I think this is a really good suggestion - thank you very much :)  I'm one of the designers on the team, just taken a quick look and have a couple of suggested amends and a question.

Can I just confirm please if the option appears on private _and_ public rooms?  It's useful for both but I couldn't be sure from the gifs if its present for both.  Thanks.

**Some design proposed changes**
I would like to propose some changes that might make the experience a bit easier for users:

1. Copy changes (in the context menu)
- ~~`Invite to this room`~~` Invite People` (I've used a capital P just because the other things in the context menu use capital case for each word, for consistency)
2. Copy changes (in the invite dialog)
- ~~`Invite to this room`~~` Invite to room name`

**Reasons**

1. I realise 'invite to this room' is the text used in the member list panel but its quite wordy for a small context menu which could scale to have another 4 options one day - the shorter all the text is (especially when translated), the easier it is to scan.  I personally think 'invite to this room' generally is too much anyway but its really highlighted in a small context menu that's trying to be small.

2. Some people, depending on the room's sensitivity, will feel re-assured by seeing the room's name in the title of the invite dialog.    There will be some users that would open, see dialog, close, re-open just to be sure it's the right room. 

Thanks, let me know if you have any issues/questions/follow up for me.

**Comment by jaiwanth-v** [comment]
Hi @niquewoodhouse, yes the invite option appears both on private and public rooms. The proposed changes look good. I'll make them soon.

**Review by niquewoodhouse — APPROVED** [review]

File: src/components/views/rooms/RoomTile.tsx [hunk 3]
```diff
@@ -453,6 +464,16 @@ export default class RoomTile extends React.PureComponent<IProps, IState> {
...
+            const inRoom = this.props.room.getMyMembership() === "join";
+            const userId = MatrixClientPeg.get().getUserId();
+            let canInvite = inRoom;
+            const powerLevels = this.props.room.currentState
+                .getStateEvents("m.room.power_levels", "")
+                ?.getContent();
+            const me = this.props.room.getMember(userId);
+            if (powerLevels && me && powerLevels.invite > me.powerLevel) {
+                canInvite = false;
+            }
```
Comment (t3chguy) [inline src/components/views/rooms/RoomTile.tsx:476]: This got stuck in review so long that I ended up adding a `canInvite()` method on a room :D
Could we use it here pelase

**Review by t3chguy — APPROVED** [review]
LGTM otherwise (again)

File: src/components/views/rooms/RoomTile.tsx (hunk 3 above)
Comment (jaiwanth-v) [reply src/components/views/rooms/RoomTile.tsx:476]: Sure

**Comment by t3chguy** [comment]
Thanks for your contribution!

**Pull request merged**